hooks/gen_docs.py

```diff
@@ -1,10 +1,169 @@
 """Generates the whole 'generatable' KPOps documentation"""
+import os
 import subprocess
+from pathlib import Path
 
 from hooks import PATH_ROOT
+from kpops.cli.registry import _find_classes
+from kpops.components.base_components.pipeline_component import PipelineComponent
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_PIPELINE_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+INHERITANCE_REF: dict = {
+    component.get_component_type(): component.__base__.get_component_type()
+    for component in KPOPS_COMPONENTS
+}
+COMPONENTS_FIELDS = {
+    component.get_component_type(): component.__fields__.keys()
+    for component in KPOPS_COMPONENTS
+}
+SECTIONS_ORDER = [
+    "type",
+    "name",
+    "namespace",
+    "app",
+    "from_",
+    "to",
+    "prefix",
+    "repo_config",
+    "version",
+    "resetter_values",
+    "offset_topic",
+]
+
+#####################
+# EXAMPLES          #
+#####################
+
+
+def get_sections(
```

Comment:
After looking at the PR I think one improvement we can do is to move the dependencies as a static file where we define the possible sections of each component. In that way we can save time and reprocessing (like sorting, etc)

Reply:
I did that, but added a check if the dependencies are up to date